real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/sensors/SensorManager.java

```diff
@@ -18,97 +19,84 @@
 import com.roboclub.robobuggy.fauxNodes.FauxIMUNode;
 import com.roboclub.robobuggy.fauxNodes.FauxNode;
 import com.roboclub.robobuggy.fauxNodes.FauxSteeringNode;
-//import com.roboclub.robobuggy.nodes.EncoderNode;
 import com.roboclub.robobuggy.nodes.GpsNode;
 import com.roboclub.robobuggy.nodes.ImuNode;
+import com.roboclub.robobuggy.nodes.RBSMNode;
 import com.roboclub.robobuggy.nodes.RealNodeEnum;
-//import com.roboclub.robobuggy.nodes.SteeringNode;
-import com.roboclub.robobuggy.ros.Node;
 import com.roboclub.robobuggy.ros.SensorChannel;
+import com.roboclub.robobuggy.serial.SerialNode;
 import com.roboclub.robobuggy.simulation.FauxRunner;
 
 public class SensorManager {
 	
 	//Need to manage real, simulated, and calculated sensors
 	
-	private LinkedHashMap<String, Node> realSensors;
+	private LinkedHashMap<String, SerialNode> realSensors;
 	private LinkedHashMap<String, LinkedHashMap<SensorChannel, FauxNode>> simulatedSensors;
 	private LinkedHashMap<String, BaseCalculatedNode> calculatedSensors;
 	private static SensorManager sm;
 	
 	private SensorManager() {
-		realSensors = new LinkedHashMap<String, Node>();
+		realSensors = new LinkedHashMap<String, SerialNode>();
 		simulatedSensors = new LinkedHashMap<String, LinkedHashMap<SensorChannel, FauxNode>>();
 		calculatedSensors = new LinkedHashMap<String, BaseCalculatedNode>();	
 	}
 
 	// Open a serial port
-	private static SerialPort connect(String portName) throws Exception
-    {
-        CommPortIdentifier portIdentifier = CommPortIdentifier.getPortIdentifier(portName);
-        if ( portIdentifier.isCurrentlyOwned() ) {
-            System.out.println("Error: Port is currently in use");
-            return null;
-        } else {
-        	//TODO fix this so that it is not potato 
-            CommPort commPort = portIdentifier.open("potato", 2000);
-            
-            if ( commPort instanceof SerialPort ) {
-                SerialPort serialPort = (SerialPort) commPort;
-                serialPort.setSerialPortParams(57600,SerialPort.DATABITS_8,SerialPort.STOPBITS_1,SerialPort.PARITY_NONE);
-                return serialPort;
-            } else {
-                System.out.println("Error: Only serial ports are handled by this example.");
-            }
-        }
+	// Returns null if unable to connect, otherwise SerialPort
+	private static SerialPort connect(String portName, int baudRate) {
+		try {
+	        CommPortIdentifier portIdentifier = CommPortIdentifier.getPortIdentifier(portName);
+	        
+	        if ( portIdentifier.isCurrentlyOwned() ) {
+	        	System.err.println("Error: Port currently in use");
+	        } else { 
+	            CommPort commPort = portIdentifier.open(portName, 2000);
```

Comment:
remove the magic number, make it a static final int defined at the top of the file instead